diff --git a/src/main.rs b/src/main.rs
index 995b915..7ed8167 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -54,7 +54,7 @@ fn init_folders() {
 fn init_internationalization() -> Result<(), Box<dyn std::error::Error>> {
     if let Ok(loc) = gettextrs::TextDomain::new("de.schmidhuberj.tubefeeder")
         .locale_category(gettextrs::LocaleCategory::LcAll)
-        .prepend("./po")
+        .prepend("/app/myshare")
         .init()
     {
         log::debug!(
@@ -66,7 +66,7 @@ fn init_internationalization() -> Result<(), Box<dyn std::error::Error>> {
         gettextrs::TextDomain::new("de.schmidhuberj.tubefeeder")
             .locale_category(gettextrs::LocaleCategory::LcAll)
             .locale("en_US")
-            .prepend("./po")
+            .prepend("/app/myshare")
             .init()?;
     }
     Ok(())
