diff --git a/src/downloader.rs b/src/downloader.rs
index c0f03b4..9e6ef76 100644
--- a/src/downloader.rs
+++ b/src/downloader.rs
@@ -32,6 +32,6 @@ pub fn download<
     let download_dir = std::env::var("XDG_DOWNLOAD_DIR")
         .unwrap_or("$HOME/Downloads/%(title)s-%(id)s.%(ext)s".to_string());
     let downloader_str =
-        std::env::var("DOWNLOADER").unwrap_or(format!("youtube-dl --output {}", download_dir));
+        std::env::var("DOWNLOADER").unwrap_or(format!("yt-dlp --output {}", download_dir));
     open_with(url, downloader_str, callback);
 }
