diff --git a/src/player.rs b/src/player.rs
index ccef408..695ae5a 100644
--- a/src/player.rs
+++ b/src/player.rs
@@ -30,7 +30,7 @@ pub fn play(video: AnyVideo) {
         log::debug!("Playing video with title: {}", video.title());
         log::debug!("Playing video with url: {}", video.url());
         video.play();
-        let player_str = std::env::var("PLAYER").unwrap_or("mpv --ytdl".to_string());
+        let player_str = std::env::var("PLAYER").unwrap_or("mpv --ytdl --script-opts=ytdl_hook-ytdl_path=yt-dlp".to_string());
 
         let mut player_iter = player_str.split(" ");
         let player = player_iter.next().unwrap_or("mpv");
