diff --git a/src/player.rs b/src/player.rs
index d846a0f..03a4819 100644
--- a/src/player.rs
+++ b/src/player.rs
@@ -31,7 +31,8 @@ pub fn play<
     callback: F,
 ) {
     log::debug!("Playing video with url: {}", url);
-    let player_str = std::env::var("PLAYER").unwrap_or("mpv --ytdl".to_string());
+    let player_str = std::env::var("PLAYER")
+        .unwrap_or("mpv --ytdl --script-opts=ytdl_hook-ytdl_path=yt-dlp".to_string());
     open_with(url, player_str, callback);
 }
